cmd/ebrelayer/txs/relayToCosmos.go

```diff
@@ -33,64 +30,41 @@ func RelayToCosmos(cdc *codec.Codec, moniker, password string, claims []types.Et
 
 	for _, claim := range claims {
 		// Packages the claim as a Tendermint message
-		msg := ethbridge.NewMsgCreateEthBridgeClaim(claim)
+		msg := types.NewMsgCreateEthBridgeClaim(claim)
 
-		err := msg.ValidateBasic()
-		if err != nil {
+		e := msg.ValidateBasic()
+		if e != nil {
 			sugaredLogger.Errorw("failed to get message from claim.",
-				errorMessageKey, err.Error())
+				errorMessageKey, e.Error())
 			continue
 		} else {
-			messages = append(messages, msg)
+			messages = append(messages, &msg)
 		}
 	}
 
-	// Prepare tx
-	txBldr, err := utils.PrepareTxBuilder(txBldr, cliCtx)
-	if err != nil {
-		sugaredLogger.Errorw("failed to get tx builder.",
-			errorMessageKey, err.Error(),
-			"transactionBuilder", txBldr)
-		return err
-	}
-
 	sugaredLogger.Infow("relay sequenceNumber from builder.",
-		"nextSequenceNumber", txBldr.Sequence())
+		"nextSequenceNumber", nextSequenceNumber)
 
 	// If we start to control sequence
 	if nextSequenceNumber > 0 {
-		txBldr.WithSequence(nextSequenceNumber)
 		sugaredLogger.Infow("txBldr.WithSequence(nextSequenceNumber) passed")
 	}
 
 	log.Println("building and signing")
-	// Build and sign the transaction
-	txBytes, err := txBldr.BuildAndSign(moniker, password, messages)
-	if err != nil {
-		sugaredLogger.Errorw("failed to sign transaction.",
-			errorMessageKey, err.Error())
-		return err
-	}
 
 	log.Println("built tx, now broadcasting")
 	// Broadcast to a Tendermint node
-	res, err := cliCtx.BroadcastTxAsync(txBytes)
+	err := tx.GenerateOrBroadcastTxCLI(cliCtx, nil, messages...)
 	if err != nil {
 		sugaredLogger.Errorw("failed to broadcast tx to sifchain.",
 			errorMessageKey, err.Error())
 		return err
 	}
 	log.Println("Broadcasted tx without error")
 
-	if err = cliCtx.PrintOutput(res); err != nil {
-		sugaredLogger.Errorw("failed to print out result.",
-			errorMessageKey, err.Error())
-		return err
-	}
-
 	// start to control sequence number after first successful tx
 	if nextSequenceNumber == 0 {
```

Comment:
I'd like to see some explanation of how nextSequenceNumber is being used now.  This was the source of many subtle problems, and the changes here are a bit confusing.  (I don't see how it's being used at all any more - am I missing something?)

Reply:
agree, the nextSequenceNumber is useless if txBldr.WithSequence was removed.

Reply:
@banshee Do you want to create a ticket to make some changes on this point, so that we can get this PR approved and merged in? It's an improvement that get's unit tests passing, so can open another PR for fine tuning from your team's side. cc @juniuszhou